sklearn/mixture/bayesian_mixture.py

```diff
@@ -0,0 +1,687 @@
+
+    Parameters
+    ----------
+    dirichlet_concentration : array-like, shape (n_samples,)
+        The parameters values of the Dirichlet distribution.
+
+    Returns
+    -------
+    log_dirichlet_norm : float
+        The log normalization of the Dirichlet distribution.
+    """
+    return (gammaln(np.sum(dirichlet_concentration)) -
+            np.sum(gammaln(dirichlet_concentration)))
+
+
+def _log_wishart_norm(freedom_degrees, log_det_precisions_chol, n_features):
+    """Compute the log of the Wishart distribution normalization term.
+
+    Parameters
+    ----------
+    freedom_degrees : array-like, shape (n_components,)
+        The parameters values of the Whishart distribution.
+
+    log_det_precision_chol : array-like, shape (n_components,)
+        The determinant of the cholesky decomposition of the precision matrix.
+
+    n_features : int
+        The number of features.
+
+    Return
+    ------
+    log_wishart_norm : array-like, shape (n_components,)
+        The log normalization of the Wishart distribution.
+    """
+    # To simplify the computation we have removed the np.log(np.pi) term
+    return -(freedom_degrees * log_det_precisions_chol +
+             freedom_degrees * n_features * .5 * np.log(2.) +
+             np.sum(gammaln(.5 * (freedom_degrees -
+                                  np.arange(0, n_features)[:,
+                                                           np.newaxis])), 0))
+
+
+class BayesianGaussianMixture(BaseMixture):
+    """Variational estimation of a Gaussian mixture.
+
+    Representation of a variational inference for a Bayesian Gaussian mixture
+    model probability distribution. This class allows to do inference of an
+    approximate posterior distribution over the parameters of a Gaussian
+    mixture distribution. The number of components can be inferred from the
+    data.
+
+    Read more in the :ref:`User Guide <bgmm>`.
+
+    Parameters
+    ----------
+    n_components: int, defaults to 1.
+        The number of mixture components.
+
+    covariance_type : {'full', 'tied', 'diag', 'spherical'},
+        defaults to 'full'.
+        String describing the type of covariance parameters to use.
+        Must be one of::
+        'full' (each component has its own general covariance matrix).
+        'tied' (all components share the same general covariance matrix),
+        'diag' (each component has its own diagonal covariance matrix),
+        'spherical' (each component has its own single variance),
+
+    tol : float, defaults to 1e-3.
+        The convergence threshold. EM iterations will stop when the
+        lower bound average gain on the likelihood (of the training data with
+        respect to the model) is below this threshold.
+
+    max_iter : int, default to 100.
+        The number of EM iterations to perform.
+
+    n_init : int, default to 1.
+        The number of initializations to perform. The best results is kept.
```

Comment:
The best results is kept.

can you clarify? eg

The result with the highest likelihood / something?